ITISFoundation/osparc-simcore
# ✨Autoscaling: 1st draft on auto-scaling computational clusters

## What do these changes do?
This allows the autoscaling service to optionally connect to a dask-scheduler to find out whether it is in need of dask-workers to complete pending jobs.

For context:
- a user starts a job (though publicAPI, or through the osparc UI) that should run on on-demand clusters
- the *clusters-keeper* service creates a primary machine on AWS EC2 where a dask-scheduler and autoscaling shall run [**upcoming PRs**],
- the autoscaling service connects to the dask-scheduler and monitors for *unrunnable jobs*, if it finds some it will try to create a machine that accomodates the job required resources, [**THIS PR**]
- the newly created machine will auto-start a dask-sidecar, that auto-connects to the dask-scheduler as soon as the machine is available, [**THIS PR**]
- the job runs, and the results are fetched by the director-v2
- the autoscaling then removes the machine if it is unused for some time, [**THIS PR**]

### details
- [x] adds dask distributed dependencies
- [x] removes old outdated script.py from sandbox
- [x] adds computational banner on start in the autoscaling logs
- [x] new ENV ```DASK_MONITORING_URL``` for use when autoscaling is to be set in computational mode (NOTE: DASK_MONITORING_URL and NODE_MONITORING settings cannot be both set)
- [x] new ENV ```EC2_INSTANCES_NAME_PREFIX``` defaults to "autoscaling" allows to prefix machines created by the autoscaling service



## Related issue/s
- requires https://github.com/ITISFoundation/osparc-simcore/pull/4709
- requires https://github.com/ITISFoundation/osparc-simcore/pull/4874



## How to test
```
make test-dev-unit
```
otherwise manual testing see ```tests/manual/README.md```



## DevOps Checklist
- new optional ENV ```EC2_INSTANCES_NAME_PREFIX```, which default to "autoscaling" to keep backwards compatibility with dynamic services autoscaling.
